app/services/ElasticsearchPlugin.scala

```diff
@@ -807,6 +808,28 @@ class ElasticsearchPlugin(application: Application) extends Plugin {
         builder.startObject().startObject("exists").field("field", key).endObject().endObject()
       })
 
+      // If user is superadmin or there is no user, no filters applied
```

Comment:
this comment seems strange, this makes it sound like if a user is not logged in they will get same list back as superadmin.

Reply:
At the moment all the endpoints for search require login, so right now user would have no path to this code. If in future someone made /search public, then without a user yes we are missing a piece in elasticsearch to filter by public. 

we could potentially get a list of spaces that are public and apply those as the filter for that case, but since the code can't currently be reached i did not do so.

Reply:
ok, will work on adding the public spaces filter for people not logged in.

Reply:
I added some basic handling in there but will need to be verified during testing #83.